vassal-app/src/main/java/VASSAL/build/module/properties/GlobalTranslatableMessages.java

```diff
@@ -0,0 +1,177 @@
+ */
+public class GlobalTranslatableMessages extends AbstractConfigurable implements TranslatableStringContainer, ToolBarComponent, PropertySource {
+  private final TemporaryToolBar tempToolbar = new TemporaryToolBar();
+  private PropertySource propertySource;
+  private final Map<String, TranslatableString> initialValues = new HashMap<>();
+  private TranslatableStringContainer parent;
+
+  @Override
+  public String[] getAttributeDescriptions() {
+    return new String[0];
+  }
+
+  @Override
+  public Class<?>[] getAttributeTypes() {
+    return new Class<?>[0];
+  }
+
+  @Override
+  public String[] getAttributeNames() {
+    return new String[0];
+  }
+
+  @Override
+  public Configurer getConfigurer() {
+    return null;
+  }
+
+  public static String getConfigureTypeName() {
+    return Resources.getString("Editor.GlobalTranslatableMessages.component_type");
+  }
+
+  @Override
+  public void setAttribute(String key, Object value) {
+  }
+
+  @Override
+  public String getAttributeValueString(String key) {
+    return null;
+  }
+
+  @Override
+  public void removeFrom(Buildable parent) {
+  }
+
+  @Override
+  public HelpFile getHelpFile() {
+    return HelpFile.getReferenceManualPage("GlobalTranslatableMessages.html"); //NON-NLS
+  }
+
+  @Override
+  public Class<?>[] getAllowableConfigureComponents() {
+    return new Class<?>[] { GlobalTranslatableMessage.class };
+  }
+
+  @Override
+  public void addTo(Buildable parent) {
+    this.parent = (TranslatableStringContainer) parent;
+
+    for (final Map.Entry<String, TranslatableString> e : initialValues.entrySet()) {
+      this.parent.addTranslatableString(e.getKey(), e.getValue());
+    }
+
+    tempToolbar.setDelegate((ToolBarComponent) parent);
+    propertySource = (PropertySource) parent;
+  }
+
+  @Override
+  public void addTranslatableString(String key, TranslatableString p) {
+    if (parent == null) {
+      initialValues.put(key, p);
+    }
+    else {
+      parent.addTranslatableString(key, p);
+    }
+  }
+
+  @Override
+  public TranslatableString removeTranslatableString(String key) {
+    if (parent == null) {
+      return initialValues.remove(key);
+    }
+    else {
+      return parent.removeTranslatableString(key);
+    }
+  }
+
+  @Override
+  public JToolBar getToolBar() {
+    return tempToolbar.getToolBar();
+  }
+
+  @Override
+  public Object getProperty(Object key) {
+    return propertySource == null ? null : propertySource.getProperty(key);
+  }
+
+  @Override
+  public Object getLocalizedProperty(Object key) {
+    return propertySource == null ? null : propertySource.getLocalizedProperty(key);
+  }
+
+  @Override
+  public GlobalTranslatableMessage getTranslatableString(String name) {
+    GlobalTranslatableMessage property = null;
+    for (final GlobalTranslatableMessage prop : getComponentsOf(GlobalTranslatableMessage.class)) {
+      if (prop.getConfigureName().equals(name)) {
+        property = prop;
```

Comment:
Should we either break out of the loop here or just return?

Reply:
done